src/test/fuzz/torcontrol.cpp

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) 2020 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <test/fuzz/FuzzedDataProvider.h>
+#include <test/fuzz/fuzz.h>
+#include <test/fuzz/util.h>
+#include <torcontrol.h>
+
+#include <cstdint>
+#include <string>
+#include <vector>
+
+class DummyTorControlConnection : public TorControlConnection
+{
+public:
+    DummyTorControlConnection() : TorControlConnection{nullptr}
+    {
+    }
+
+    bool Connect(const std::string&, const ConnectionCB&, const ConnectionCB&)
+    {
+        return true;
+    }
+
+    void Disconnect()
+    {
+    }
+
+    bool Command(const std::string&, const ReplyHandlerCB&)
+    {
+        return true;
+    }
+};
+
+void initialize()
+{
+    InitializeFuzzingContext();
+}
+
+void test_one_input(const std::vector<uint8_t>& buffer)
+{
+    FuzzedDataProvider fuzzed_data_provider{buffer.data(), buffer.size()};
+
+    TorController tor_controller;
+    while (fuzzed_data_provider.ConsumeBool()) {
+        TorControlReply tor_control_reply;
+        if (fuzzed_data_provider.ConsumeBool()) {
+            tor_control_reply.code = 250;
+        } else if (fuzzed_data_provider.ConsumeBool()) {
+            tor_control_reply.code = 510;
+        } else {
+            tor_control_reply.code = fuzzed_data_provider.ConsumeIntegral<int>();
+        }
+        tor_control_reply.lines = ConsumeRandomLengthStringVector(fuzzed_data_provider);
+        if (tor_control_reply.lines.empty()) {
+            break;
+        }
+        DummyTorControlConnection dummy_tor_control_connection;
```

Comment:
Can't this be replaced by `TorControlConnection tor_control_connection{nullptr};` and then you can get rid of the `DummyTorControlConnection` class?

Reply:
We don't want `TorControlConnection::{Connect,Disconnect,Command}` to be called, right? :)  My intention was to add `DummyTorControlConnection` to make sure `{Connect,Disconnect,Command}` return immediately (and thus be guaranteed to be safe regardless of the state of things).

Reply:
Yes I think that's the right approach, I think my coverage confused me again as it showed `DummyTorControlConnection` as unused.